Java/C#: add `isUninteresting` to `ExternalApi` characteristic predicate

jcogs33 opened the pull request:
### Description
This PR adds the `isUninteresting` predicate into the characteristic predicate for `ExternalApi`.

### Consideration
- Should I add a public `isParameterlessConstructor` predicate to the `Callable` class (where both `ExternalApi` and `DataFlowTargetApi` [#11624](https://github.com/github/codeql/pull/11624) could use it) instead of keeping it as a private predicate in `ExternalApi.qll`?
- Does a change like this need a DCA run?
- Is this change minor enough that I can add the `no-change-note-required` label?

(cc @michaelnebel)

michaelnebel commented on csharp/ql/src/Telemetry/ExternalApi.qll:
```diff
@@ -24,14 +24,28 @@ class TestLibrary extends RefType {
+
+/** Holds if the given callable is not worth supporting. */
+private predicate isUninteresting(DotNet::Callable c) {
+  isTestLibrary(c) or isParameterlessConstructor(c)
```
Inline isTestLibrary and isParameterlessConstructor.

jcogs33 replied:
Dumb question, but just to be sure: Is this a style suggestion to move `isParameterlessConstructor(c)` underneath `isTestLibrary(c)`? Or are you asking me to use [pragma[inline]](https://codeql.github.com/docs/ql-language-reference/annotations/#pragma-inline) for these?

jcogs33 replied:
ah, that makes a lot more sense, thanks! Updated in https://github.com/github/codeql/pull/11631/commits/a889cc4f6555e907debb4f23aefa631a43280c97.

michaelnebel commented on csharp/ql/src/Telemetry/ExternalApi.qll:
```diff
@@ -24,14 +24,28 @@ class TestLibrary extends RefType {
 
+/** Holds if the given callable is a constructor without parameters. */
+private predicate isParameterlessConstructor(DotNet::Callable c) {
+  c instanceof Constructor and c.getNumberOfParameters() = 0
```
Also you also suggests it makes sense to define a predicate we can re-use as it is also needed on the other PR (please see my comment there - we could consider to add `isParameterless()` to the constructor class and re-use it).

michaelnebel reviewed (changes requested):
Yes, this looks good - I have added some suggestions.

The changes on this PR are semantics preserving, so maybe it is good idea to run DCA with the `mad-queries.qls` suite for java `commons-io` and C# `dotnet/dotnet`. The DCA run should show that there are no changes in the generated models.

jcogs33 commented:
Thanks for the review! I've addressed your suggestions in https://github.com/github/codeql/pull/11631/commits/cde93a39cd576ed1b907e7d4356a6bd3a4ce480a and https://github.com/github/codeql/pull/11631/commits/123fe5edfe073ffa12242322410a03d51311f98e, and I've started DCA runs.

michaelnebel reviewed (approved):
Great work Jami!
